src/sentry/api/endpoints/organization_missing_org_members.py

```diff
@@ -17,66 +18,78 @@
 from sentry.api.api_publish_status import ApiPublishStatus
 from sentry.api.base import region_silo_endpoint
 from sentry.api.bases.organization import OrganizationEndpoint, OrganizationPermission
-from sentry.api.serializers import Serializer, serialize
 from sentry.constants import ObjectStatus
 from sentry.integrations.base import IntegrationFeatures
-from sentry.models import Repository
-from sentry.models.commit import Commit
-from sentry.models.commitauthor import CommitAuthor
 from sentry.models.organization import Organization
-from sentry.search.utils import tokenize_query
 from sentry.services.hybrid_cloud.integration import integration_service
 
-if TYPE_CHECKING:
-    from django_stubs_ext import WithAnnotations
-
-
 filtered_email_domains = {
     "gmail.com",
     "icloud.com",
     "hotmail.com",
     "outlook.com",
     "noreply.github.com",
+    "localhost",
 }
 
 filtered_characters = {"+"}
 
 
-class MissingOrgMemberSerializer(Serializer):
-    def serialize(self, obj, attrs, user, **kwargs):
-        return {"email": obj.email, "externalId": obj.external_id, "commitCount": obj.commit__count}
-
-
 class MissingMembersPermission(OrganizationPermission):
     scope_map = {"GET": ["org:write"]}
 
 
-def _get_missing_organization_members(
-    organization: Organization, provider: str, integration_ids: Sequence[int]
-) -> QuerySet[WithAnnotations[CommitAuthor]]:
-    member_emails = set(
-        organization.member_set.exclude(email=None).values_list("email", flat=True)
-    ) | set(organization.member_set.exclude(user_email=None).values_list("user_email", flat=True))
-
-    nonmember_authors = CommitAuthor.objects.filter(organization_id=organization.id).exclude(
-        Q(email__in=member_emails) | Q(external_id=None)
-    )
-
-    org_repos = Repository.objects.filter(
-        provider="integrations:" + provider,
-        organization_id=organization.id,
-        integration_id__in=integration_ids,
-    ).values_list("id", flat=True)
-
-    recent_commits = Commit.objects.filter(
-        repository_id__in=set(org_repos), date_added__gte=timezone.now() - timedelta(days=30)
-    ).values_list("id", flat=True)
-
-    return (
-        nonmember_authors.filter(commit__id__in=recent_commits)
-        .annotate(Count("commit"))
-        .order_by("-commit__count")
-    )
+def _get_missing_organization_members_query(
```

Comment:
worth looking into if its better to write raw SQL or trying to use `SubQuery` django stuff etc. e.g. https://github.com/getsentry/sentry/blob/a97d7d66b4fabffdb2b7ddbed31deaed48b4301e/src/sentry/monitors/endpoints/organization_monitor_index.py#L123